litebird_sim/coordinates.py

```diff
@@ -1,9 +1,121 @@
 # -*- encoding: utf-8 -*-
+import numpy as np
 
 from astropy.coordinates import BarycentricMeanEcliptic
+from astropy.coordinates import SkyCoord
+
+from enum import Enum
 
 """The coordinate system used by the framework"""
 DEFAULT_COORDINATE_SYSTEM = BarycentricMeanEcliptic()
 
 """The time scale used by the framework"""
 DEFAULT_TIME_SCALE = "tdb"
+
+ECL_TO_GAL_ROT_MATRIX = (
+    SkyCoord(
+        x=[1.0, 0.0, 0.0],
+        y=[0.0, 1.0, 0.0],
+        z=[0.0, 0.0, 1.0],
+        frame=DEFAULT_COORDINATE_SYSTEM.name,
+        representation_type="cartesian",
+    )
+    .transform_to("galactic")
+    .data.to_cartesian()
+    .get_xyz()
+    .value
+)
+
+CoordinateSystem = Enum("CoordinateSystem", ["Ecliptic", "Galactic"])
+
+
+def ang2vec(theta, phi):
+    """Transform a direction theta,phi to a unit vector.
+
+    Parameters
+    ----------
+    theta : float, scalar or array-like
+      The angle theta (scalar or shape (N,))
+    phi : float, scalar or array-like
+      The angle phi (scalar or shape (N,)).
+
+    Returns
+    -------
+    vec : array
+      The vector(s) corresponding to given angles, shape is (3,) or (3, N).
+
+    See Also
+    --------
+    https://github.com/healpy/healpy/blob/main/healpy/rotator.py#L657
+    """
+    ct, st, cp, sp = np.cos(theta), np.sin(theta), np.cos(phi), np.sin(phi)
+    vec = np.empty((3, ct.size), np.float64)
+    vec[0, :] = st * cp
+    vec[1, :] = st * sp
+    vec[2, :] = ct
+    return vec.squeeze()
+
+
+def vec2ang(vx, vy, vz):
+    """Transform a vector to angle given by theta,phi.
+
+    Parameters
+    ----------
+    vx : float, scalar or array-like
+      The x component of the vector (scalar or shape (N,))
+    vy : float, scalar or array-like, optional
+      The y component of the vector (scalar or shape (N,))
+    vz : float, scalar or array-like, optional
+      The z component of the vector (scalar or shape (N,))
+
+    Returns
+    -------
+    angles : float, array
+      The angles in radiants in an array of shape (2, N)
+
+    See Also
+    --------
+    https://github.com/healpy/healpy/blob/main/healpy/rotator.py#L610
+    """
+
+    r = np.sqrt(vx ** 2 + vy ** 2 + vz ** 2)
+    ang = np.empty((2, r.size))
+    ang[0, :] = np.arccos(vz / r)
```

Comment:
I suggest to use something like `ang[0] = np.arctan2(np.sqrt(vx**2 + vy**2),vz)`, which is more accurate near the poles and does not break in presence of numerical noise. See for example https://gitlab.mpcdf.mpg.de/mtr/ducc/-/blob/ducc0/src/ducc0/math/pointing.cc#L46

Reply:
My personal rule of thumb is to _never_ use `arcsin` or `arccos` anywhere, they are causing too much trouble because of their lmited definition range and their infinite derivatives near the borders. There is always a better solution using `arctan2`...